src/Microsoft.Identity.Web.AgentIdentities/AgentUserIdentityMsalAddIn.cs

```diff
@@ -0,0 +1,88 @@
+                            serviceProvider.GetRequiredService<IOptionsMonitor<MicrosoftIdentityApplicationOptions>>();
+
+                        // Get the FIC token for the agent application.
+                        string authenticationScheme = authenticationSchemeInformationProvider.GetEffectiveAuthenticationScheme(options.AuthenticationOptionsName);
+                        ITokenAcquirer tokenAcquirer = tokenAcquirerFactory.GetTokenAcquirer(authenticationScheme);
+                        ITokenAcquirer agentApplicationTokenAcquirer = tokenAcquirerFactory.GetTokenAcquirer();
+                        AcquireTokenResult aaFic = await agentApplicationTokenAcquirer.GetFicTokenAsync(new() { FmiPath = agentIdentity }); // Uses the regular client credentials
+                        string? clientAssertion = aaFic.AccessToken;
+
+                        // Get the FIC token for the agent identity.
+                        MicrosoftIdentityApplicationOptions microsoftIdentityApplicationOptions = optionsMonitor.Get(authenticationScheme);
+                        ITokenAcquirer agentIdentityTokenAcquirer = tokenAcquirerFactory.GetTokenAcquirer(new MicrosoftIdentityApplicationOptions
+                        {
+                            ClientId = agentIdentity,
+                            Instance = microsoftIdentityApplicationOptions.Instance,
+                            Authority = microsoftIdentityApplicationOptions.Authority,
+                            TenantId = microsoftIdentityApplicationOptions.TenantId
+                        });
+                        AcquireTokenResult aidFic = await agentIdentityTokenAcquirer.GetFicTokenAsync(clientAssertion: clientAssertion); // Uses the agent identity
+                        string? userFicAssertion = aidFic.AccessToken;
+
+                        // Already in the request:
+                        // - client_id = agentIdentity;
+
+                        // User FIC parameters
+                        request.BodyParameters["client_assertion_type"] = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+                        request.BodyParameters["client_assertion"] = clientAssertion;
+                        request.BodyParameters["username"] = username;
+                        request.BodyParameters["user_federated_identity_credential"] = userFicAssertion;
+                        request.BodyParameters["grant_type"] = "user_fic";
+                        request.BodyParameters.Remove("password");
+
+                        if (request.BodyParameters.TryGetValue("client_secret", out var secret)
+                                && secret.Equals("default", StringComparison.OrdinalIgnoreCase))
+                        {
+                            request.BodyParameters.Remove("client_secret");
+                        }
+
+                        // For the moment
+                        request.RequestUri = new Uri(request.RequestUri + "?slice=first");
```

Comment:
Can this not be in the query params?

Reply:
I tried to set ExtraQueryParaterms in the test and they do not show up as `AcquireTokenOptions` here. Not sure why. 

Maybe you can use a special options.ExtraParameters or smth? We should not put this slice param in an official release, this has caused problems in the past.